Add CHANGELOG.md (#548)
arborx/ArborX: CHANGELOG.md

- Improve performance of the brute force algorithm [\#534](https://github.com/arborx/ArborX/pull/534)
- Improve performance of the DBSCAN algorithm (new FDBSCAN-DenseBox algorithm) [\#508](https://github.com/arborx/ArborX/pull/508)
- Allow non-View input for DBSCAN [\#509](https://github.com/arborx/ArborX/pull/509)
- Install CMake package version file [\#521](https://github.com/arborx/ArborX/pull/521)
- Add a simple intersection example [\#513](https://github.com/arborx/ArborX/pull/513)
- Improve performance for the SYCL backend through the use of oneDPL for sorting [\#456](https://github.com/arborx/ArborX/pull/456)

**Fixed bugs:**

- Fix a bug in DBSCAN noise marking [\#525](https://github.com/arborx/ArborX/pull/525)

## [v1.0](https://github.com/arborx/arborx/tree/v1.0) (2021-03-13)

[Full Changelog](https://github.com/arborx/arborx/compare/v1.0-rc0...v1.0)

**New features:**

- Allow early termination of a traversal by a thread [\#427](https://github.com/arborx/ArborX/pull/427)
- Implement DBSCAN clustering algorithm [\#331](https://github.com/arborx/ArborX/pull/331)
- Implement brute-force algorithm [\#468](https://github.com/arborx/ArborX/pull/468)

> - Add initial ray-tracing support [\#414](https://github.com/arborx/ArborX/pull/414)

Author: ```suggestion
- Add initial ray-tracing support [\#414](https://github.com/arborx/ArborX/pull/414) and [\#461](https://github.com/arborx/ArborX/pull/461)
```

> 

Reviewer: - Add ray-sphere intersection algorithm [\#461](https://github.com/arborx/ArborX/pull/461)

Reviewer 2: @wjge I think it falls under initial ray tracing support. No need to go into details.

Reviewer: The CHANGELOG is also giving the users a summary on what they can do more with ArborX for this release/update.
More details may help the potential users to locate the right pieces of code/functions/example/tests through the PR. Later when Bruno's ray-triangle intersection gets merged, we can add another line like '-Add ray-triangle intersection algorithm'.

**Build changes:**

- Require CMake 3.16 [\#486](https://github.com/arborx/ArborX/pull/486)

**Enhancements:**

> - Add KOKKOS\_FUNCTION to AccessTraits::size\(\) in View specialization [\#463](https://github.com/arborx/ArborX/pull/463)

Author: ```suggestion
- Add `KOKKOS_FUNCTION` to `AccessTraits::size()` in View specialization [\#463](https://github.com/arborx/ArborX/pull/463)
```

> - Non-backward compatible: Change signature of the nearest callback [\#366](https://github.com/arborx/ArborX/pull/366)

Reviewer 3: I think that non-backward compatible changes should be in their own section. It's easier for user to know if they need to change their code.

Author: Fixed in [`8b565ab` (#548)](https://github.com/arborx/ArborX/pull/548/commits/8b565abc43d398de3d26d7be96157cb992150bba).

- Add `query()` free function [\#425](https://github.com/arborx/ArborX/pull/425)
- Improve performance for the HIP backend through the use of rocThrust for sorting [\#424](https://github.com/arborx/ArborX/pull/424)
- Support for SYCL and OpenMPTarget [\#422](https://github.com/arborx/ArborX/pull/422)

## [v1.0-rc0](https://github.com/arborx/arborx/tree/v1.0-rc0) (2020-10-03)

[Full Changelog](https://github.com/arborx/arborx/compare/v0.9-beta...v1.0-rc0)

**New features:**

> - New BVH::query\(\) overload that only takes predicates and callback [\#329](https://github.com/arborx/ArborX/pull/329)

Author: ```suggestion
- New `BVH::query()` overload that only takes predicates and callback [\#329](https://github.com/arborx/ArborX/pull/329)
```

**Enhancements:**

> - Implement stackless tree traversal using escape index \(ropes\) [\#364](https://github.com/arborx/ArborX/pull/364)

Author: ```suggestion
- Implement stackless tree traversal using escape index (ropes) [\#364](https://github.com/arborx/ArborX/pull/364)
```

- Add support for Kokkos HIP backend [\#236](https://github.com/arborx/ArborX/pull/236)
- Ensure that all kernels and memory allocations are prefixed with `ArborX::` [\#362](https://github.com/arborx/ArborX/issues/362) and [\#380](https://github.com/arborx/ArborX/pull/380)
- Improve performance of knn traversal [\#357](https://github.com/arborx/ArborX/pull/357)
- Add new query overloads for the distributed tree [\#356](https://github.com/arborx/ArborX/pull/356)
- Increase performance of the BVH construction [\#350](https://github.com/arborx/ArborX/pull/350)
- Allow non device type template parameter for output views in query\(\) on distributed trees [\#349](https://github.com/arborx/ArborX/pull/349)

**Fixed bugs:**

- Fix double free when making copies of a distributed tree [\#369](https://github.com/arborx/ArborX/pull/369)

> - Resolve duplicate Details::toBufferStatus\(int\) symbol error downstream [\#360](https://github.com/arborx/ArborX/pull/360)

Reviewer 2: ```suggestion
- Resolve duplicate `Details::toBufferStatus\(int\)` symbol error downstream [\#360](https://github.com/arborx/ArborX/pull/360)
```

> - Resolve duplicate `Details::toBufferStatus\(int\)` symbol error downstream [\#360](https://github.com/arborx/ArborX/pull/360)

Reviewer 2: ```suggestion
- Resolve duplicate `Details::toBufferStatus(int)` symbol error downstream [\#360](https://github.com/arborx/ArborX/pull/360)
```

- Make predicates sorting optional [\#243](https://github.com/arborx/ArborX/pull/243)
- Use user-provided execution space instances [\#250](https://github.com/arborx/ArborX/pull/250)
- Implement algorithm for finding strongly connected components (halo finder) [\#237](https://github.com/arborx/ArborX/pull/237)
- Store bounding boxes in single-precision floating-point format [\#235](https://github.com/arborx/ArborX/pull/235)
- Allow usage of CUDA-aware MPI [\#162](https://github.com/arborx/ArborX/pull/162)

**Build changes:**

- Require Kokkos 3.1 [\#268](https://github.com/arborx/ArborX/pull/268)
- Require C++14 standard when building ArborX [\#226](https://github.com/arborx/ArborX/pull/226)
- Require Kokkos CMake build [\#93](https://github.com/arborx/ArborX/pull/93)

**Enhancements:**

- Template BVH on memory space [\#251](https://github.com/arborx/ArborX/pull/251)
- Add example for callbacks and lift requirement for tagging inline [\#325](https://github.com/arborx/ArborX/pull/325)
- Enable building against Trilinos' Kokkos installation [\#156](https://github.com/arborx/ArborX/pull/156)
- Add access traits CUDA example [\#107](https://github.com/arborx/ArborX/pull/107)

> - Let BVH::bounds\(\) be a KOKKOS\_FUNCTION [\#326](https://github.com/arborx/ArborX/pull/326)

Reviewer 2: ```suggestion
- Let `BVH::bounds()` be a `KOKKOS_FUNCTION` [\#326](https://github.com/arborx/ArborX/pull/326)
```

- Improve performance of the radius search [\#306](https://github.com/arborx/ArborX/pull/306)
- Improve performance of the kNN search [\#308](https://github.com/arborx/ArborX/pull/308)

> - Deprecate `Traits::Access` in favor of `AccessTraits` [\#300](https://github.com/arborx/ArborX/pull/300)

Reviewer 3: In version 1.0-rc0, deprecated features are in their own sections. We should be consistent.

Author: Fixed in [`8b565ab` (#548)](https://github.com/arborx/ArborX/pull/548/commits/8b565abc43d398de3d26d7be96157cb992150bba).

- Retain the original path to Kokkos [\#287](https://github.com/arborx/ArborX/pull/287)
- Disable tests, examples, benchmarks by default [\#284](https://github.com/arborx/ArborX/pull/284)
- Template distributed search tree on the memory space [\#260](https://github.com/arborx/ArborX/pull/260)
- Enable predicates access traits in distributed search tree [\#196](https://github.com/arborx/ArborX/pull/196)
- Set default build type to RelWithDebInfo [\#188](https://github.com/arborx/ArborX/pull/188)
- Remove all fences [\#150](https://github.com/arborx/ArborX/pull/150)
- Improve performance for sorting with CUDA by using Thrust [\#147](https://github.com/arborx/ArborX/pull/147)

> - Improve compilation error messages produced by BVH::query\(\) [\#279](https://github.com/arborx/ArborX/pulls/279)

Reviewer 2: ```suggestion
- Improve compilation error messages produced by `BVH::query()` [\#279](https://github.com/arborx/ArborX/pulls/279)
```

**Fixed bugs:**

> - Fix ambiguity in queryDispatch\(\) overload resolution [\#293](https://github.com/arborx/ArborX/pull/293)

Reviewer 2: ```suggestion
- Fix ambiguity in `queryDispatch()` overload resolution [\#293](https://github.com/arborx/ArborX/pull/293)
```

- Properly update hash in version file when building from subdirs [\#266](https://github.com/arborx/ArborX/pull/266)
- Avoid second pass for radius search when the results are empty [\#240](https://github.com/arborx/ArborX/pull/240)

> - Avoid more compiler warnings for nvcc\_wrapper [\#185](https://github.com/arborx/ArborX/pull/185)

Reviewer 2: ```suggestion
- Avoid more compiler warnings for `nvcc_wrapper` [\#185](https://github.com/arborx/ArborX/pull/185)
```

> - Fix segfault in Distributor [\#296](https://github.com/arborx/ArborX/pulls/296)

Reviewer 2: ```suggestion
- Fix segfault in `Distributor` [\#296](https://github.com/arborx/ArborX/pulls/296)
```

- Allow non device type template parameter for output views in BVH::query\(\) [\#271](https://github.com/arborx/ArborX/pull/271)

## [v0.8-beta2](https://github.com/arborx/arborx/tree/v0.8-beta2) (2019-10-10)

[Full Changelog](https://github.com/arborx/arborx/compare/97bbec21cc92dd2b4bd3a68c52a230b4c3c4643c...v0.8-beta2)

**New features:**

- Provide access traits for predicates [\#117](https://github.com/arborx/ArborX/pull/117)
- Add `version()` and `gitCommitHash()` [\#61](https://github.com/arborx/ArborX/pull/61)

**Build changes:**

- Replace TriBITS build system with "raw" CMake [\#14](https://github.com/arborx/ArborX/pull/14)
- Rename `ArborX_ENABLE_XYZ` options to `ARBORX_ENABLE_XYZ` [\#99](https://github.com/arborx/ArborX/pull/99)
- Provide an option to disable tests and examples [\#31](https://github.com/arborx/ArborX/pull/31)
- Make MPI dependency optional [\#42](https://github.com/arborx/ArborX/pull/42)

**Enhancements:**

> - Use MPI\_Comm\_dup to separate ArborX comm context from user's [\#135](https://github.com/arborx/ArborX/pull/135)

Reviewer 2: ```suggestion
- Use `MPI_Comm_dup` to separate ArborX comm context from user's [\#135](https://github.com/arborx/ArborX/pull/135)
```

- Add CMake option for enabling benchmarks [\#138](https://github.com/arborx/ArborX/pull/138)

> - Add intersects\(Point, Box\) [\#122](https://github.com/arborx/ArborX/pull/122)

Reviewer 2: ```suggestion
- Add intersection of a `Point` with `Box` [\#122](https://github.com/arborx/ArborX/pull/122)
```

- Install CMake package version file [\#521](https://github.com/arborx/ArborX/pull/521)
- Add a simple intersection example [\#513](https://github.com/arborx/ArborX/pull/513)
- Improve performance for the SYCL backend through the use of oneDPL for sorting [\#456](https://github.com/arborx/ArborX/pull/456)

**Fixed bugs:**

- Fix a bug in DBSCAN noise marking [\#525](https://github.com/arborx/ArborX/pull/525)

## [v1.0](https://github.com/arborx/arborx/tree/v1.0) (2021-03-13)

[Full Changelog](https://github.com/arborx/arborx/compare/v1.0-rc0...v1.0)

**New features:**

- Allow early termination of a traversal by a thread [\#427](https://github.com/arborx/ArborX/pull/427)
- Implement DBSCAN clustering algorithm [\#331](https://github.com/arborx/ArborX/pull/331)
- Implement brute-force algorithm [\#468](https://github.com/arborx/ArborX/pull/468)
- Add ray-tracing example [\#414](https://github.com/arborx/ArborX/pull/414)

**Build changes:**

- Require CMake 3.16 [\#486](https://github.com/arborx/ArborX/pull/486)

**Enhancements:**

- Add KOKKOS\_FUNCTION to AccessTraits::size\(\) in View specialization [\#463](https://github.com/arborx/ArborX/pull/463)

> - Allow running BVH benchmark with SYCL and OpenMPTarget explicitly [\#455](https://github.com/arborx/ArborX/pull/455)

Reviewer 4: ```suggestion
```

Reviewer 2: I consider this user-visible.

- Non-backward compatible: Change signature of the nearest callback [\#366](https://github.com/arborx/ArborX/pull/366)

> - Add a free function to execute queries [\#425](https://github.com/arborx/ArborX/pull/425)

Reviewer 4: ```suggestion
- Add `query()` free function [\#425](https://github.com/arborx/ArborX/pull/425)
```

- Improve performance for the HIP backend through the use of rocThrust for sorting [\#424](https://github.com/arborx/ArborX/pull/424)
- Support for SYCL and OpenMPTarget [\#422](https://github.com/arborx/ArborX/pull/422)

## [v1.0-rc0](https://github.com/arborx/arborx/tree/v1.0-rc0) (2020-10-03)

[Full Changelog](https://github.com/arborx/arborx/compare/v0.9-beta...v1.0-rc0)

**New features:**

- New BVH::query\(\) overload that only takes predicates and callback [\#329](https://github.com/arborx/ArborX/pull/329)

**Implemented enhancements:**

- Implement stackless tree traversal using escape index \(ropes\) [\#364](https://github.com/arborx/ArborX/pull/364)

> - Enable CI for HIP [\#236](https://github.com/arborx/ArborX/pull/236)

Reviewer 4: ```suggestion
- Add support for Kokkos HIP backend [\#236](https://github.com/arborx/ArborX/pull/236)
```

**Build changes:**

- Replace TriBITS build system with "raw" CMake [\#14](https://github.com/arborx/ArborX/pull/14)
- Rename `ArborX_ENABLE_XYZ` options to `ARBORX_ENABLE_XYZ` [\#99](https://github.com/arborx/ArborX/pull/99)
- Provide an option to disable tests and examples [\#31](https://github.com/arborx/ArborX/pull/31)
- Make MPI dependency optional [\#42](https://github.com/arborx/ArborX/pull/42)

**Enhancements:**

- Use MPI\_Comm\_dup to separate ArborX comm context from user's [\#135](https://github.com/arborx/ArborX/pull/135)
- Add CMake option for enabling benchmarks [\#138](https://github.com/arborx/ArborX/pull/138)
- Add intersects\(Point, Box\) [\#122](https://github.com/arborx/ArborX/pull/122)
- Mark BVH::{size,empty,bounds} as noexcept [\#114](https://github.com/arborx/ArborX/pull/114)
- Improve error messages in BVH constructor and BVH::query\(\) [\#113](https://github.com/arborx/ArborX/pull/113)
- Make data private in Point [\#100](https://github.com/arborx/ArborX/pull/100)
- Constexpr geometric primitives and algorithms [\#97](https://github.com/arborx/ArborX/pull/97)
- Add CUDA+clang CI test [\#95](https://github.com/arborx/ArborX/pull/95)

> - Test installation of ArborX [\#48](https://github.com/arborx/ArborX/pull/48)

Reviewer 4: ```suggestion
```

[Full Changelog](https://github.com/arborx/arborx/compare/v1.0-rc0...v1.0)

**New features:**

- Allow early termination of a traversal by a thread [\#427](https://github.com/arborx/ArborX/pull/427)
- Implement DBSCAN clustering algorithm [\#331](https://github.com/arborx/ArborX/pull/331)
- Implement brute-force algorithm [\#468](https://github.com/arborx/ArborX/pull/468)
- Add initial ray-tracing support [\#414](https://github.com/arborx/ArborX/pull/414) and [\#461](https://github.com/arborx/ArborX/pull/461)

**Build changes:**

- Require CMake 3.16 [\#486](https://github.com/arborx/ArborX/pull/486)

**Enhancements:**

- Add `KOKKOS_FUNCTION` to `AccessTraits::size()` in View specialization [\#463](https://github.com/arborx/ArborX/pull/463)
- Change signature of the nearest callback [\#366](https://github.com/arborx/ArborX/pull/366)
- Add `query()` free function [\#425](https://github.com/arborx/ArborX/pull/425)
- Improve performance for the HIP backend through the use of rocThrust for sorting [\#424](https://github.com/arborx/ArborX/pull/424)
- Support for SYCL and OpenMPTarget [\#422](https://github.com/arborx/ArborX/pull/422)

> **Incompatibilities**

Reviewer 2: ```suggestion
**Backward incompatible changes:**
```
Just a suggestion, not blocking.

# Changelog

## [v1.1](https://github.com/arborx/arborx/tree/v1.1) (2021-09-23)

[Full Changelog](https://github.com/arborx/arborx/compare/v1.0...v1.1)

**New features:**

- Allow k-DOP as a bounding volume in BVH [\#386](https://github.com/arborx/ArborX/pull/386)

**Enhancements:**

- Store a dimension value in a DBSCAN datafile [\#523](https://github.com/arborx/ArborX/pull/523)

> - Provide computation of a Sphere-Box distance [\#495](https://github.com/arborx/ArborX/pull/495)
> - Provide Box-Box distance for the Nearest query [\#494](https://github.com/arborx/ArborX/pull/494)

Reviewer 4: ```suggestion
- Enable knn queries with spheres and boxes [\#495](https://github.com/arborx/ArborX/pull/495) and [\#494](https://github.com/arborx/ArborX/pull/494)
```

> - Use double to normalize Ray direction [\#535](https://github.com/arborx/ArborX/pull/535)

Reviewer 4: ```suggestion
- Use double-precision floating-points to normalize `Ray` direction [\#535](https://github.com/arborx/ArborX/pull/535)
```

- Improve performance of the brute force algorithm [\#534](https://github.com/arborx/ArborX/pull/534)
- Improve performance of the DBSCAN algorithm (new FDBSCAN-DenseBox algorithm) [\#508](https://github.com/arborx/ArborX/pull/508)
- Allow non-View input for DBSCAN [\#509](https://github.com/arborx/ArborX/pull/509)

> - Install ArborXConfigVersion.cmake [\#521](https://github.com/arborx/ArborX/pull/521)

Reviewer 4: ```suggestion
- Install CMake package version file [\#521](https://github.com/arborx/ArborX/pull/521)
```

- Add a simple intersection example [\#513](https://github.com/arborx/ArborX/pull/513)
- Improve performance for the SYCL backend through the use of oneDPL for sorting [\#456](https://github.com/arborx/ArborX/pull/456)

**Fixed bugs:**

- Fix a bug in DBSCAN noise marking [\#525](https://github.com/arborx/ArborX/pull/525)

## [v1.0](https://github.com/arborx/arborx/tree/v1.0) (2021-03-13)

[Full Changelog](https://github.com/arborx/arborx/compare/v1.0-rc0...v1.0)

**New features:**

> - Allow early termination of a traversal by a thread [\#427](https://github.com/arborx/ArborX/pull/427)

Reviewer 4: Should we say it is "experimental"?

Author: I don't think that's crucial.

- Implement DBSCAN clustering algorithm [\#331](https://github.com/arborx/ArborX/pull/331)
- Implement brute-force algorithm [\#468](https://github.com/arborx/ArborX/pull/468)

> - Implement simple ray-tracing [\#414](https://github.com/arborx/ArborX/pull/414)

Reviewer 4: ```suggestion
- Add ray-tracing example [\#414](https://github.com/arborx/ArborX/pull/414)
```

Reviewer 2: We did not simply add an example, we also added `Ray` class and intersections with it. I would like to revert this change.

Reviewer 4: Everything is covered by our "adding initial ray tracing support"

Reviewer 2: I don't see "Adding initial ray tracing support" anywhere in the changelog.

**Build changes:**

- Require CMake 3.16 [\#486](https://github.com/arborx/ArborX/pull/486)

**Enhancements:**

- Add KOKKOS\_FUNCTION to AccessTraits::size\(\) in View specialization [\#463](https://github.com/arborx/ArborX/pull/463)
- Allow running BVH benchmark with SYCL and OpenMPTarget explicitly [\#455](https://github.com/arborx/ArborX/pull/455)

> - Change signature of the nearest callback [\#366](https://github.com/arborx/ArborX/pull/366)

Reviewer 4: Maybe we should say this is backward incompatible or breaking change

Author: ```suggestion
- Non-backward compatible: Change signature of the nearest callback [\#366](https://github.com/arborx/ArborX/pull/366)
```

> - Add a free function constructing CRS graph [\#425](https://github.com/arborx/ArborX/pull/425)

Reviewer 4: ```suggestion
```

Author: I think we should keep this.

Reviewer 4: Adding free function `query()` sure, CRS graph mention no

Author: ```suggestion
- Add a free function to execute queries [\#425](https://github.com/arborx/ArborX/pull/425)
```

- Improve performance for the HIP backend through the use of rocThrust for sorting [\#424](https://github.com/arborx/ArborX/pull/424)

> - Add CI for SYCL [\#423](https://github.com/arborx/ArborX/pull/423)

Reviewer 4: I don't think we should mention CI

Author: I think it's important to mention that we support `SYCL`.
```suggestion
- Support for SYCL and OpenMPTarget [\#422](https://github.com/arborx/ArborX/pull/422)
```

Reviewer 4: OK with changes being suggested

## [v1.0-rc0](https://github.com/arborx/arborx/tree/v1.0-rc0) (2020-10-03)

[Full Changelog](https://github.com/arborx/arborx/compare/v0.9-beta...v1.0-rc0)

**New features:**

- New BVH::query\(\) overload that only takes predicates and callback [\#329](https://github.com/arborx/ArborX/pull/329)

**Implemented enhancements:**

> - Change naming of ArborX labels [\#362](https://github.com/arborx/ArborX/issues/362)

Reviewer 4: Combine with 380 and say "ensure all kernels and memory allocation are prefixed with `ArborX::`"

> - Change naming of ArborX labels [\#362](https://github.com/arborx/ArborX/issues/362)
> - Implement stackless tree traversal using escape index \(ropes\) [\#364](https://github.com/arborx/ArborX/pull/364)
> - Enable CI for HIP [\#236](https://github.com/arborx/ArborX/pull/236)
> - Ensure that all kernels from BVH are properly marked [\#380](https://github.com/arborx/ArborX/pull/380)

Author: ```suggestion
- Implement stackless tree traversal using escape index \(ropes\) [\#364](https://github.com/arborx/ArborX/pull/364)
- Enable CI for HIP [\#236](https://github.com/arborx/ArborX/pull/236)
- Ensure that all kernels and memory allocations are prefixed with `ArborX::` [\#362](https://github.com/arborx/ArborX/issues/362) and [\#380](https://github.com/arborx/ArborX/pull/380)
```

- Improve performance of knn traversal [\#357](https://github.com/arborx/ArborX/pull/357)
- Add new query overloads for the distributed tree [\#356](https://github.com/arborx/ArborX/pull/356)
- Increase performance of the BVH construction [\#350](https://github.com/arborx/ArborX/pull/350)
- Allow non device type template parameter for output views in query\(\) on distributed trees [\#349](https://github.com/arborx/ArborX/pull/349)

**Fixed bugs:**

- Fix double free when making copies of a distributed tree [\#369](https://github.com/arborx/ArborX/pull/369)
- Resolve duplicate Details::toBufferStatus\(int\) symbol error downstream [\#360](https://github.com/arborx/ArborX/pull/360)
- Fix narrowing conversion warnings [\#343](https://github.com/arborx/ArborX/pull/343)

**Deprecations:**

> - Deprecate DistributedSearchTree [\#396](https://github.com/arborx/ArborX/pull/396)

Reviewer 4: ```suggestion
- Deprecate `DistributedSearchTree` in favor of `DistributedTree` [\#396](https://github.com/arborx/ArborX/pull/396)
```

## [v0.9-beta](https://github.com/arborx/arborx/tree/v0.9-beta) (2020-06-10)

[Full Changelog](https://github.com/arborx/arborx/compare/v0.8-beta2...v0.9-beta)

**New features:**

- Enable user-defined callback in BVH search [\#166](https://github.com/arborx/ArborX/pull/166)
- Make predicates sorting optional [\#243](https://github.com/arborx/ArborX/pull/243)

> - Use user-provided execution space [\#250](https://github.com/arborx/ArborX/pull/250)

Reviewer 4: ```suggestion
- Use user-provided execution space instances [\#250](https://github.com/arborx/ArborX/pull/250)
```

- Implement algorithm for finding strongly connected components (halo finder) [\#237](https://github.com/arborx/ArborX/pull/237)

> - Use single precision floating point numbers for data [\#235](https://github.com/arborx/ArborX/pull/235)

Reviewer 4: ```suggestion
- Store bounding boxes in single-precision floating-point format [\#235](https://github.com/arborx/ArborX/pull/235)
```

- Allow usage of CUDA-aware MPI [\#162](https://github.com/arborx/ArborX/pull/162)

**Build changes:**

- Require Kokkos 3.1 [\#268](https://github.com/arborx/ArborX/pull/268)
- Require C++14 standard when building ArborX [\#226](https://github.com/arborx/ArborX/pull/226)
- Require Kokkos CMake build [\#93](https://github.com/arborx/ArborX/pull/93)

**Implemented enhancements:**

- Template BVH on memory space [\#251](https://github.com/arborx/ArborX/pull/251)
- Add example for callbacks and lift requirement for tagging inline [\#325](https://github.com/arborx/ArborX/pull/325)
- Enable building against Trilinos' Kokkos installation [\#156](https://github.com/arborx/ArborX/pull/156)
- Add access traits CUDA example [\#107](https://github.com/arborx/ArborX/pull/107)
- Let BVH::bounds\(\) be a KOKKOS\_FUNCTION [\#326](https://github.com/arborx/ArborX/pull/326)

> - Drop QUIET option when finding Kokkos [\#310](https://github.com/arborx/ArborX/pull/310)

Reviewer 4: ```suggestion
```

- Improve performance of the radius search [\#306](https://github.com/arborx/ArborX/pull/306)
- Improve performance of the kNN search [\#308](https://github.com/arborx/ArborX/pull/308)

> - Rename Traits::Access to AccessTraits [\#300](https://github.com/arborx/ArborX/pull/300)

Reviewer 4: ```suggestion
- Deprecate `Traits::Access` in favor of `AccessTraits` [\#300](https://github.com/arborx/ArborX/pull/300)
```

- Retain the original path to Kokkos [\#287](https://github.com/arborx/ArborX/pull/287)
- Disable tests, examples, benchmarks by default [\#284](https://github.com/arborx/ArborX/pull/284)

> - Improve error messages for inline callbacks [\#279](https://github.com/arborx/ArborX/pull/279)

Reviewer 4: Duplicate of L119.  Remove.

Author: ```suggestion
```

- Template distributed search tree on the memory space [\#260](https://github.com/arborx/ArborX/pull/260)
- Enable predicates access traits in distributed search tree [\#196](https://github.com/arborx/ArborX/pull/196)
- Set default build type to RelWithDebInfo [\#188](https://github.com/arborx/ArborX/pull/188)
- Remove all fences [\#150](https://github.com/arborx/ArborX/pull/150)

> - Use Thrust for GPU sort [\#147](https://github.com/arborx/ArborX/pull/147)

Reviewer 4: Maybe say perf improvement

Author: ```suggestion
- Improve performance for sorting with CUDA by using Thrust [\#147](https://github.com/arborx/ArborX/pull/147)
```

- Improve compilation error messages produced by BVH::query\(\) [\#279](https://github.com/arborx/ArborX/pulls/279)

**Fixed bugs:**

> - Use Clang workaround for Thrust regarless of the version [\#288](https://github.com/arborx/ArborX/pull/288)

Reviewer 4: ```suggestion
```

- Fix ambiguity in queryDispatch\(\) overload resolution [\#293](https://github.com/arborx/ArborX/pull/293)
- Properly update hash in version file when building from subdirs [\#266](https://github.com/arborx/ArborX/pull/266)
- Avoid second pass for radius search when the results are empty [\#240](https://github.com/arborx/ArborX/pull/240)
- Avoid more compiler warnings for nvcc\_wrapper [\#185](https://github.com/arborx/ArborX/pull/185)

> - Suppress Thrust warnings [\#178](https://github.com/arborx/ArborX/pull/178)

Reviewer 4: ```suggestion
```

- Fix segfault in Distributor [\#296](https://github.com/arborx/ArborX/pulls/296)
- Allow non device type template parameter for output views in BVH::query\(\) [\#271](https://github.com/arborx/ArborX/pull/271)

## [v0.8-beta2](https://github.com/arborx/arborx/tree/v0.8-beta2) (2019-10-10)

[Full Changelog](https://github.com/arborx/arborx/compare/97bbec21cc92dd2b4bd3a68c52a230b4c3c4643c...v0.8-beta2)

**New features:**

- Provide access traits for predicates [\#117](https://github.com/arborx/ArborX/pull/117)

> - Add version\(\) and gitCommitHash\(\) [\#61](https://github.com/arborx/ArborX/pull/61)

Reviewer 4: ```suggestion
- Add `version()` and `gitCommitHash()` [\#61](https://github.com/arborx/ArborX/pull/61)
```

**Build changes:**

> - Replace TRIBITS build system with CMake [\#14](https://github.com/arborx/ArborX/pull/14)

Reviewer 4: ```suggestion
- Replace TriBITS build system with "raw" CMake [\#14](https://github.com/arborx/ArborX/pull/14)
```

> - Replace ArborX\_ENABLE with ARBORX\_ENABLE [\#99](https://github.com/arborx/ArborX/pull/99)

Reviewer 4: ```suggestion
- Rename `ArborX_ENABLE_XYZ` options to `ARBORX_ENABLE_XYZ` [\#99](https://github.com/arborx/ArborX/pull/99)
```

**Enhancements:**

- Use MPI\_Comm\_dup to separate ArborX comm context from user's [\#135](https://github.com/arborx/ArborX/pull/135)
- Add CMake option for enabling benchmarks [\#138](https://github.com/arborx/ArborX/pull/138)
- Add intersects\(Point, Box\) [\#122](https://github.com/arborx/ArborX/pull/122)
- Mark BVH::{size,empty,bounds} as noexcept [\#114](https://github.com/arborx/ArborX/pull/114)
- Improve error messages in BVH constructor and BVH::query\(\) [\#113](https://github.com/arborx/ArborX/pull/113)
- Make data private in Point [\#100](https://github.com/arborx/ArborX/pull/100)
- Constexpr geometric primitives and algorithms [\#97](https://github.com/arborx/ArborX/pull/97)
- Add CUDA+clang CI test [\#95](https://github.com/arborx/ArborX/pull/95)
- Make sure the recorded git hash is recent [\#79](https://github.com/arborx/ArborX/pull/79)
- Test installation of ArborX [\#48](https://github.com/arborx/ArborX/pull/48)
- Relax CudaUVM requirement [\#24](https://github.com/arborx/ArborX/pull/24)
- Find Boost in subdirectories that actually require it [\#22](https://github.com/arborx/ArborX/pull/22)

**Fixed bugs:**

> - Fix parsing options in the ArborX benchmark [\#54](https://github.com/arborx/ArborX/pulls/54)
> - Fix compilation against Kokkos development branch [\#40](https://github.com/arborx/ArborX/pulls/40)

Reviewer 4: ```suggestion
```

Reviewer 2: I would revert this, I find it useful, especially the second one.

Reviewer 4: It makes little sense, especially for the initial release

- Optimize communication within the same rank [\#134](https://github.com/arborx/ArborX/pull/134)

> - Fix for distributed searches with large count of results per que… [\#129](https://github.com/arborx/ArborX/pull/129)

Reviewer 4: ```suggestion
- Fix for distributed searches with large count of results per query [\#129](https://github.com/arborx/ArborX/pull/129)
```

> - Fix Kokkos\_DEVICES spelling [\#35](https://github.com/arborx/ArborX/pull/35)

Reviewer 4: ```suggestion
```

> - Add missing STATUS keyword in message [\#98](https://github.com/arborx/ArborX/pull/98)

Reviewer 4: ```suggestion
```

> - Fix memory management in CmdLineArgs [\#66](https://github.com/arborx/ArborX/pull/66)
> - Fix memory violation when processing cmd line args in the driver [\#63](https://github.com/arborx/ArborX/pull/63)
> - Fix unused variable warnings [\#25](https://github.com/arborx/ArborX/pull/25)

Reviewer 4: ```suggestion
```

Reviewer 2: This is debatable. I consider command line options in the examples and benchmarks to be user-visible, and thus reportable. What's your view on it?

Reviewer 4: Prior to initial release, irrelevant IMO

> 
> **Deprecations:**
> 
> - Deprecate old aliases to generate predicates [\#119](https://github.com/arborx/ArborX/pull/119)
> - Disable deprecated code [\#40](https://github.com/arborx/ArborX/pull/40)

Reviewer 4: ```suggestion
```

Reviewer 2: Why? I think deprecations should be in the changelog.

Reviewer 4: Deprecation as compared to what.  Also the 2nd one is quit using deprecated Kokkos functionalities

Reviewer 2: OK with removing the second one. First one is deprecation to the previous version.
